app/server/lib/ExternalStorage.ts

```diff
@@ -67,8 +67,27 @@ export interface ExternalStorage {
  * E.g. this could convert "<docId>" to "v1/<docId>.grist"
  */
 export class KeyMappedExternalStorage implements ExternalStorage {
+  public uploadStream: ExternalStorage['uploadStream'];
+  public downloadStream: ExternalStorage['downloadStream'];
+  public removeAllWithPrefix: ExternalStorage['removeAllWithPrefix'];
+
   constructor(private _ext: ExternalStorage,
               private _map: (key: string) => string) {
+    if (_ext.uploadStream !== undefined) {
```

Comment:
(to anyone reviewing: this is all fine, dealing with some methods being optional and trying to keep them so, but let's not get hung up on code golf trying to make this clean since the methods are going to become non-optional soon once native s3 and azure backends are updated, so all this will evaporate)